rpcs3/rpcs3qt/trophy_manager_dialog.cpp

```diff
@@ -641,62 +642,57 @@ void trophy_manager_dialog::ShowContextMenu(const QPoint& loc)
 	menu->exec(globalPos);
 }
 
-void trophy_manager_dialog::StartTrophyLoadThread()
-{
-	auto progressDialog = new QProgressDialog(
-		tr("Loading trophy data, please wait..."), tr("Cancel"), 0, 1, this,
-		Qt::Dialog | Qt::WindowTitleHint | Qt::CustomizeWindowHint);
-	progressDialog->setWindowTitle(tr("Loading trophies"));
-	connect(progressDialog, &QProgressDialog::canceled, [this]()
-	{
-		TrophyThreadState expected = TrophyThreadState::RUNNING;
-		m_thread_state.compare_exchange_strong(expected, TrophyThreadState::CLOSING);
-		this->close(); // It's pointless to show an empty window
-	});
-	progressDialog->show();
-
-	auto trophyThread = new trophy_manager_dialog::trophy_load_thread(this);
-	connect(trophyThread, &QThread::finished, trophyThread, &QThread::deleteLater);
-	connect(trophyThread, &QThread::finished, progressDialog, &QProgressDialog::deleteLater);
-	connect(trophyThread, &trophy_manager_dialog::trophy_load_thread::TotalCountChanged, progressDialog, &QProgressDialog::setMaximum);
-	connect(trophyThread, &trophy_manager_dialog::trophy_load_thread::ProcessedCountChanged, progressDialog, &QProgressDialog::setValue);
-	connect(trophyThread, &trophy_manager_dialog::trophy_load_thread::FinishedSuccessfully, [this]() { RepaintUI(true); });
-	m_thread_state = TrophyThreadState::RUNNING;
-	trophyThread->start();
-}
-
-void trophy_manager_dialog::trophy_load_thread::run()
+void trophy_manager_dialog::StartTrophyLoadThreads()
 {
-	m_manager->m_trophies_db.clear();
+	m_trophies_db.clear();
 
-	QDir trophy_dir(qstr(vfs::get(m_manager->m_trophy_dir)));
+	QDir trophy_dir(qstr(vfs::get(m_trophy_dir)));
 	const auto folder_list = trophy_dir.entryList(QDir::Dirs | QDir::NoDotAndDotDot);
 	const int count = folder_list.count();
-	Q_EMIT TotalCountChanged(count);
 
-	for (int i = 0; m_manager->m_thread_state == TrophyThreadState::RUNNING && i < count; i++)
+	if (count <= 0)
+	{
+		RepaintUI(true);
+		return;
+	}
+
+	QList<int> indices;
```

Comment:
Ditto

Reply:
i is used as well as compared as an int in the following code. so size_t is not making any sense here